src/task_monitor.rs

```diff
@@ -239,6 +254,24 @@ impl TaskMonitor {
 
         handles.push(insert_identities_handle);
 
+        // // Delete identities task
+        let delete_identites = DeleteIdentities::new(
+            self.database.clone(),
+            self.identity_manager.clone(),
+            self.tree_state.get_latest_tree(),
+            self.deletion_time_interval,
+            self.min_batch_deletion_size,
+            wake_up_notify,
```

Comment:
This `Notify` gets triggered when something is inserted via the api. Can you make sure it's also triggered on deletion or add a new notify just for deletion?